crates/symbolicator-service/src/services/symbolication/sourcemap.rs

```diff
@@ -85,44 +123,96 @@ fn js_processing_symbolicate_frame(
         }
     };
 
-    let mut result = JsProcessingSymbolicatedFrame {
-        status: JsProcessingFrameStatus::Symbolicated,
-        raw: JsProcessingRawFrame {
-            function: Some(fold_function_name(&function_name)),
-            filename: frame.filename.clone(),
-            abs_path: token.file_name().unwrap_or("<unknown>").to_string(),
-            // TODO: Decide where to do off-by-1 calculations
-            lineno: Some(token.line().saturating_add(1)),
-            colno: Some(token.column().saturating_add(1)),
-            pre_context: vec![],
-            context_line: None,
-            post_context: vec![],
-        },
-    };
+    result.raw.function = Some(fold_function_name(&function_name));
+    result.raw.abs_path = token.file_name().unwrap_or("<unknown>").to_string();
+    result.raw.lineno = Some(token.line().saturating_add(1));
+    result.raw.colno = Some(token.column().saturating_add(1));
 
     if let Some(file) = token.file() {
-        result.raw.filename = file.name().map(|name| name.to_string());
+        result.raw.filename = file.name().map(|f| f.to_string());
+
+        if file.source().is_some() {
+            apply_source_context_from_sourcemap_cache(&mut result, file).await;
```

Comment:
as a followup, we should maybe batch these together as a separate step.

Reply:
That would be possible as well I guess